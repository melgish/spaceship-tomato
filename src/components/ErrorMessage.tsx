import type { FieldError, Merge } from 'react-hook-form'

type Props = Readonly<{
  error?: FieldError | Merge<FieldError, (FieldError | undefined)[]>
}>

export default function ErrorMessage({ error }: Props) {
  if (Array.isArray(error)) {
    error = error.flat()[0]
  }

  if (!error) {
    return null
  }

  return <div className="error text-rose-800">{error?.message ?? 'Unexpected'}</div>
}
