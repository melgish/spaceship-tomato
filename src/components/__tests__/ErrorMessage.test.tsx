import { describe, expect, it, vi } from 'vitest'
import { render } from '@testing-library/react'
import ErrorMessage from '@/components/ErrorMessage'

const data = { message: 'Error' }

describe('when there is no error', () => {
  it('should not render content', () => {
    const { container } = render(<ErrorMessage />)

    expect(container).toMatchSnapshot()
  })
})

describe('when there is an array of errors', () => {
  it('should not render the first item', () => {
    const { container } = render(<ErrorMessage error={[data] as any} />)

    expect(container).toMatchSnapshot()
  })
})

describe('when error has no message', () => {
  it('should not render unexpected', () => {
    const { container } = render(<ErrorMessage error={{} as any} />)

    expect(container).toMatchSnapshot()
  })
})
