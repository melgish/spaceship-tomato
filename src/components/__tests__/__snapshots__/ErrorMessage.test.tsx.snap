// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when error has no message > should not render unexpected 1`] = `
<div>
  <div
    class="error text-rose-800"
  >
    Unexpected
  </div>
</div>
`;

exports[`when there is an array of errors > should not render the first item 1`] = `
<div>
  <div
    class="error text-rose-800"
  >
    Error
  </div>
</div>
`;

exports[`when there is no error > should not render content 1`] = `<div />`;
